internal/controllers/clusterrequest_controller.go

```diff
@@ -833,6 +1070,200 @@ func (r *ClusterRequestReconciler) handleFinalizer(
 	return ctrl.Result{}, false, nil
 }
 
+func (t *clusterRequestReconcilerTask) waitForHardwareData(ctx context.Context,
```

Comment:
Could you add a comment describing what this function is doing? Slightly hard to follow...but it's not just waiting (deciding if ready) but final goal is to update NodePool yeah? Would nice split into two e.g `IsPluginManagerReady` and `UpdateNodePool` 

Also shouldnt we bubble up these errors and show it in status?

Reply:
A comment is added and it is split into two functions.   It is to be determined  whether to display the internal errors in the status.